framework/encode/vulkan_capture_layer_settings.h

```diff
@@ -0,0 +1,103 @@
+** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+** FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+** DEALINGS IN THE SOFTWARE.
+*/
+
+#ifndef GFXRECON_FRAMEWORK_ENCODE_VULKAN_LAYER_SETTINGS_H
+#define GFXRECON_FRAMEWORK_ENCODE_VULKAN_LAYER_SETTINGS_H
+
+#include "vulkan/vulkan.h"
+
+#include "graphics/vulkan_struct_get_pnext.h"
+
+#include "encode/capture_layer_settings.h"
+#include "encode/capture_settings.h"
+
+GFXRECON_BEGIN_NAMESPACE(gfxrecon)
+GFXRECON_BEGIN_NAMESPACE(encode)
+
+CaptureLayerSettings GetVulkanCaptureLayerSettings(const VkInstanceCreateInfo* pCreateInfo)
+{
+    CaptureLayerSettings layer_settings;
+
+    if (pCreateInfo == nullptr)
+    {
+        return layer_settings;
+    }
+
+    // There can be multiple VkLayerSettingsCreateInfoEXT structures in the pNext chain.
+    const VkBaseInStructure* p_next = reinterpret_cast<const VkBaseInStructure*>(pCreateInfo);
+    while (p_next != nullptr)
+    {
+        const auto* settings_create_info = graphics::vulkan_struct_get_pnext<VkLayerSettingsCreateInfoEXT>(p_next);
+        if (settings_create_info == nullptr)
+        {
+            // No more to find
+            break;
+        }
+
+        static const char* VK_LAYER_NAME = "VK_LAYER_LUNARG_gfxreconstruct";
+        for (uint32_t i = 0; i < settings_create_info->settingCount; ++i)
+        {
+            const VkLayerSettingEXT& setting = settings_create_info->pSettings[i];
+
+            // Only add settings for the GFXReconstruct layer
+            if (setting.pLayerName != nullptr && strcmp(setting.pLayerName, VK_LAYER_NAME) == 0)
+            {
+                if (setting.pSettingName == nullptr)
+                {
+                    GFXRECON_LOG_ERROR("Setting name is null for layer '%s'", VK_LAYER_NAME);
+                    continue;
+                }
+
+                // Get capture name setting
+                if (strcmp(setting.pSettingName, CAPTURE_FILE_NAME_LOWER) == 0)
+                {
+                    if (setting.type != VK_LAYER_SETTING_TYPE_STRING_EXT)
+                    {
+                        GFXRECON_LOG_ERROR("Expected setting type for '%s' to be "
+                                           "VK_LAYER_SETTING_TYPE_STRING_EXT, but got %d",
+                                           CAPTURE_FILE_NAME_LOWER,
+                                           setting.type);
+                    }
+                    else if (setting.valueCount == 0 || setting.pValues == nullptr)
+                    {
+                        GFXRECON_LOG_ERROR("Expected non-empty value for '%s' setting, but got empty value",
+                                           CAPTURE_FILE_NAME_LOWER);
+                    }
+                    else
+                    {
+                        layer_settings.capture_filename =
+                            std::string(static_cast<const char*>(setting.pValues), setting.valueCount);
```

Comment:
Just another reminder that `pValues` is an array of null-terminated `char*` and `valueCount` is the length of the array (1)